extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheResourceProcessor.java

```diff
@@ -185,5 +202,24 @@ void build(CombinedIndexBuildItem index,
                 }
             }
         }
+
+        Map<String, Set<String>> collectedEntityToPersistenceUnits = new HashMap<>();
+        if (jpaModelPersistenceUnitMapping.isPresent()) {
+            collectedEntityToPersistenceUnits = jpaModelPersistenceUnitMapping.get().getEntityToPersistenceUnits();
+        }
+
+        for (Map.Entry<String, Set<String>> entry : collectedEntityToPersistenceUnits.entrySet()) {
+            String entityName = entry.getKey();
+            List<String> selectedPersistenceUnits = new ArrayList<>(entry.getValue());
+            boolean isPanacheEntity = modelClasses.stream().anyMatch(name -> name.equals(entityName));
+            if (selectedPersistenceUnits.size() > 1 && isPanacheEntity) {
+                throw new IllegalStateException(String.format(
+                        "PanacheEntity '%s' cannot be defined for usage in several persistence units which is not supported. The following persistence units were found: %s.",
+                        entityName, String.join(",", selectedPersistenceUnits)));
+            }
+
+            panacheEntityToPersistenceUnit.put(entityName, selectedPersistenceUnits.get(0));
+        }
+        recorder.setEntityToPersistenceUnit(panacheEntityToPersistenceUnit);
```

Comment:
Is this really needed? I would have expected the standard ORM processor to be triggered too? Or am I missing something?

Reply:
If it's necessary, I will need to make the error message consistent but I would be surprised it is.

Reply:
> Is this really needed? I would have expected the standard ORM processor to be triggered too? Or am I missing something?

Yes, this is needed. The kotlin panache extension does not make use of the hibernate panache extension. They all share a common spi module.

Reply:
Ok, I will adjust the PR then. Running the tests, I will push once they're green.